File: app/src/main/java/msc/app/embalsespuertorico/DamMoreInfoPager.java
package msc.app.embalsespuertorico;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

//Extending FragmentStatePagerAdapter
public class DamMoreInfoPager extends FragmentStatePagerAdapter {

    private int tabCount;
    public DamMoreInfoPager(FragmentManager fm, int tabCount) {
        super(fm);
        this.tabCount= tabCount;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new GetDamInfoAndShow();
            case 1:
                return new DamMoreInfoFragment();
            case 2:
                return new DamHistoryFragment();
            case 3:
                return new DamGraph();
            case 4:
                return new DamMaps();
            default:
                return null;
        }
    }
    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }
    @Override
    public int getCount() {
        return tabCount;
    }
}

File: app/src/main/java/msc/app/embalsespuertorico/DamMoreInfoPager.kt
package msc.app.embalsespuertorico

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter

//Extending FragmentStatePagerAdapter
class DamMoreInfoPager(fm: FragmentManager, private val tabCount: Int) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> GetDamInfoAndShow()
            1 -> DamMoreInfoFragment()
            2 -> DamHistoryFragment()
            3 -> DamGraph()
            4 -> DamMaps()
            else -> GetDamInfoAndShow()
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getCount(): Int {
        return tabCount
    }
}